lib/propolis/src/tasks.rs

```diff
@@ -412,3 +416,94 @@ impl Drop for CtrlHeld<'_> {
         }
     }
 }
+
+/// Holds a group of tokio task [task::JoinHandle]s to be later joined as a
+/// group when they have all concluded.
+pub struct TaskGroup(Mutex<Vec<task::JoinHandle<()>>>);
```

Comment:
This feels like it could, potentially, be a good use case for [`tokio::task::JoinSet`](https://docs.rs/tokio/latest/tokio/task/struct.JoinSet.html) rather than `FuturesUnordered`, which is a more efficient way to join a group of spawned tasks. However, it offers a somewhat different API surface; you can't add `JoinHandle`s to a `JoinSet` and must, instead, spawn the tasks on the `JoinSet` using its own `JoinSet::spawn` family of methods.[^1]

Definitely not something we have to do in this PR, just thought I'd mention it.

[^1]: This is because `JoinSet`'s optimizations rely on it being integrated with the runtime and only able to operate on spawned tasks, rather than arbitrary futures which may not be tasks.

Reply:
Yeah, I don't think it's worth the effort to change everything that uses this type in this PR, I just thought I'd mention it for future reference.